src/test/java/uk/gov/pay/adminusers/pact/ContractTestSuite.java

```diff
@@ -1,13 +1,40 @@
 package uk.gov.pay.adminusers.pact;
 
+import junit.framework.JUnit4TestAdapter;
+import junit.framework.TestSuite;
 import org.junit.runner.RunWith;
+import org.junit.runners.AllTests;
 import org.junit.runners.Suite;
+import org.slf4j.Logger;
+import org.slf4j.LoggerFactory;
 
-@RunWith(Suite.class)
+import java.util.Map;
 
-@Suite.SuiteClasses({
-        ProviderContractTest.class
-})
+import static java.lang.String.format;
+
+@RunWith(AllTests.class)
 public class ContractTestSuite {
+    
+    private static final Logger LOGGER = LoggerFactory.getLogger(ContractTestSuite.class);
+
+    private static Map<String, JUnit4TestAdapter> map = Map.of(
+            "products-ui", new JUnit4TestAdapter(ProductsUIContractTest.class),
+            "selfservice", new JUnit4TestAdapter(SelfServiceContractTest.class));
+
+    public static TestSuite suite() {
+        String consumer = System.getProperty("CONSUMER");
+        TestSuite suite = new TestSuite();
+
+        if (consumer == null || consumer.isBlank()) {
+            LOGGER.info("Running all contract tests.");
+            map.forEach((key, value) -> suite.addTest(value));
+        } else if (map.containsKey(consumer)) {
+            LOGGER.info("Running {}-connector contract tests only.", consumer);
+            suite.addTest(map.get(consumer));
+        } else {
+            throw new RuntimeException(format("Error running provider contract tests. ${CONSUMER} system property was %s.", consumer));
+        }
 
+        return suite;
+    }
```

Comment:
Reckon we could add a base class or method somewhere in pay-java-commons for this? Might be overkill though..

Reply:
Yeah sounds good